lib/vanagon/utilities.rb

```diff
@@ -281,5 +281,16 @@ def erb_file(erbfile, outfile = nil, remove_orig = false, opts = { :binding => b
       FileUtils.rm_rf erbfile if remove_orig
       outfile
     end
+
+    # Final
+    # - 1.0.0
+    # - 3.0.5.1
+    # - 4.3.0-5
+    # Nonfinal
+    # - 1.0.0-34-g1sadjf9283
+    # - 4.2.1.2344.gfdqw83rh
+    def final_version?(version)
+      !version.match(/g[a-f0-9]{7}$/)
```

Comment:
`g123456789` is a short sha that's generated based on the long sha of the commit `git rev-parse HEAD` will show you the long sha. It's always hexadecimal characters, hence the  `[a-f0-9]{7}`. The way vanagon gets that short sha is always going to be 7 characters in length. The way git reports it always has a leading `g`.